pkg/pod/status.go

```diff
@@ -217,16 +219,30 @@ func createMessageFromResults(results []v1beta1.PipelineResourceResult) (string,
 	return string(bytes), nil
 }
 
-func filterResultsAndResources(results []v1beta1.PipelineResourceResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
+func filterResultsAndResources(ctx context.Context, results []v1beta1.PipelineResourceResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
 	var taskResults []v1beta1.TaskRunResult
 	var pipelineResourceResults []v1beta1.PipelineResourceResult
 	var filteredResults []v1beta1.PipelineResourceResult
 	for _, r := range results {
 		switch r.ResultType {
 		case v1beta1.TaskRunResultType:
+			cfg := config.FromContextOrDefaults(ctx)
+			aos := v1beta1.ArrayOrString{}
+			if cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
```

Comment:
Is this check necessary here? We are already validating specifications ⬆️  https://github.com/tektoncd/pipeline/pull/4818/files#diff-2a1291b0e1080edbd9fdcd0c6e4c6bd4bfc2967e2c376815219bb571aeeae70fR30

Reply:
Yes we have validated in admission webhook, but I feel like it is no wrong to add more checks in reconciler code if possible.

Reply:
The side effect is we are bringing `context` down for no reason. This validation does not add any value and in long run, might require more cycles to understand why its implemented here.

Reply:
Yes! Another reason we validate here is that it is possible that users don't specify the result type, it is an optional field and will be set to `string` if user not specified. If the type is required mandatory it will introduce breaking changes. 
So the validation in webhook is not 100% reliable in the case that user creates a task without setting the result type.
And we cannot infer the result type in validation webhook since the results are emitted during runtime.
That's why it may make sense to validate when we get the emitted results at reconciler.